src/monitormanager.h

```diff
@@ -48,8 +49,11 @@ class MonitorManager : public IndexingObject<Monitor> {
     void unlock();
     std::string lock_number_changed();
 
+    void extractWindowStack(bool real_clients, std::function<void(Window)> addToStack);
+    void restack();
+
     int cur_monitor;
-    Stack* monitor_stack;
+    std::unique_ptr<PlainStack<Monitor*>> monitorStack_; // TODO: make this private
```

Comment:
The lifetime of `monitorStack_` exactly matches that of `MonitorManager`, so this could actually just be a plain old member instead of a unique_ptr.

I've made the change to confirm that it works, so if you agree with it, you can just cherry-pick this: https://github.com/dnnr/herbstluftwm/commit/c79b6626d589873cacc874ad5ca9861dce9775b9

Reply:
I've cherry-picked it, thanks! Originally, I wanted to avoid to include the rather long stack.h in monitormanager.h, but with the shorter plainstack.h, I think a member is better. Thanks!